cli/src/diff_util.rs

```diff
@@ -749,6 +760,108 @@ pub fn show_diff_summary(
+                DiffLineType::Added => adds += 1,
+            }
+        }
+    }
+    DiffStat {
+        path: repo_path,
+        added: adds,
+        removed: dels,
+    }
+}
+
+pub fn show_diff_stat(
+    ui: &Ui,
+    formatter: &mut dyn Formatter,
+    workspace_command: &WorkspaceCommandHelper,
+    tree_diff: TreeDiffIterator,
+) -> Result<(), CommandError> {
+    let mut stats: Vec<DiffStat> = vec![];
+    let mut max_path_length = 0;
+    let mut max_diffs = 0;
+    for (repo_path, diff) in tree_diff {
+        let path = workspace_command.format_file_path(&repo_path);
+        match diff {
+            tree::Diff::Modified(left, right) => {
+                let left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                let right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+                let stat = get_diff_stat(path, &left_content, &right_content);
+                stats.push(stat);
+            }
+            tree::Diff::Added(right) => {
+                let right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+                let stat = get_diff_stat(path, &[], &right_content);
+                stats.push(stat);
+            }
+            tree::Diff::Removed(left) => {
+                let left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                let stat = get_diff_stat(path, &left_content, &[]);
+                stats.push(stat);
+            }
+        }
+        max_path_length = max(max_path_length, repo_path.to_internal_file_string().len());
+        max_diffs = stats
+            .iter()
+            .map(|stat| stat.added + stat.removed)
+            .max()
+            .unwrap_or(0);
+    }
+
+    let display_width = usize::from(ui.term_width().unwrap_or(80)) - 4; // padding
+    let max_bar_length =
+        display_width - max_path_length - " | ".len() - max_diffs.to_string().len() - 1;
+    let factor = if max_diffs < max_bar_length {
+        1.0
+    } else {
+        max_bar_length as f64 / max_diffs as f64
+    };
+
+    formatter.with_label("diff", |formatter| {
+        let mut total_added = 0;
+        let mut total_removed = 0;
+        for stat in &stats {
+            total_added += stat.added;
+            total_removed += stat.removed;
+            let bar_added = (stat.added as f64 * factor).ceil() as usize;
+            let bar_removed = (stat.removed as f64 * factor).ceil() as usize;
+            // pad to max_path_length
+            write!(
+                formatter.labeled("hunk-header"),
+                " {:<max_path_length$} | {:>4} ",
+                stat.path,
+                stat.added + stat.removed,
+            )?;
+            write!(formatter.labeled("added"), "{}", "+".repeat(bar_added))?;
+            writeln!(formatter.labeled("removed"), "{}", "-".repeat(bar_removed))?;
+        }
+        writeln!(
+            formatter.labeled("hunk-header"),
```

Comment:
This is also not a hunk header, so let's use a different name. Maybe "stat-summary"?

Reply:
Done.